mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -335,17 +369,70 @@ class KMeans @Since("1.5.0") (
     model
   }
 
+  /**
+   * Check validity for interactions between parameters.
+   */
+  private def assertInitialModelValid(): Unit = {
+    if ($(initMode) == MLlibKMeans.K_MEANS_INITIAL_MODEL) {
+      if (isSet(initialModel)) {
+        val initialModelK = $(initialModel).parentModel.k
+        if (initialModelK != $(k)) {
+          throw new IllegalArgumentException("The initial model's cluster count = " +
+            s"$initialModelK, mismatched with k = $k.")
+        }
+      } else {
+        throw new IllegalArgumentException("Users must set param initialModel if you choose " +
+          "'initialModel' as the initialization algorithm.")
+      }
+    } else {
+      if (isSet(initialModel)) {
+        logWarning(s"Param initialModel will take no effect when initMode is $initMode.")
+      }
+    }
+  }
+
   @Since("1.5.0")
   override def transformSchema(schema: StructType): StructType = {
+    assertInitialModelValid()
```

Comment:
`transformSchema ` is called in `transform` method, and `model.transform` is called in computing the summary. I think we should fail it earlier instead of checking it in the end. Also, it's implicit that it's being checked when computing summary. We should explicitly check it. 

If we have small logic in checking, I'll have those checking code in `fit` method.